File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func removeNthFromEnd(_ head: ListNode?, _ n: Int) -> ListNode? {
        if head == nil{return nil}
        let dummy: ListNode? = ListNode(0)
        dummy?.next = head
        var l = dummy
        var r = head
        var c = n

        while c > 0{
            r = r?.next
            c -= 1
        }

        while r != nil{
            l = l?.next
            r = r?.next
        }

        l?.next = l?.next?.next
        return dummy?.next
    }

}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        head = root

        while head:
            if max(p.val, q.val) < head.val:
                head = head.left
            elif min(p.val, q.val) > head.val:
                head = head.right
            else:
                break
        
        return head